src/pages/Location/Location.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from "react";
+
+import Layout from "../../components/Layout";
+import CharacterCard from "../../components/CharacterCard";
+
+import { getLocation, getUrl } from "../../api";
+
+function makePromises(elements = []) {
+  return elements.map((url) => getUrl(url));
+}
+
+class Location extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      location: null,
+      residents: [],
+      hasLoaded: false,
+      hasError: false,
+      errorMessage: null,
+    };
+  }
+
+  componentDidMount() {
+    // console.log(this.props);  Mirar en match.params que está lo que pedimos antes por url!
+    const { match } = this.props;
+    const { locationId } = match.params;
+    // console.log(characterId);
+    this.loadLocation(locationId);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match } = this.props;
+    const { match: prevMatch } = prevProps;
+    if (match !== prevMatch) {
+      this.loadLocation(match.locationId);
+    }
+  }
+
+  async loadLocation(locationId) {
+    try {
+      const { data } = await getLocation(locationId);
+      // console.log({ data });
+
+      // eslint-disable-next-line compat/compat
+      const residentsResponse = await Promise.all(makePromises(data.residents));
+      const residents = residentsResponse.map((resident) => resident.data);
+      // console.log({ episodes });
```

Comment:
We have to get used to erasing the console.log before submitting the solution, to start with the good practices!